src/ape/managers/compilers.py

```diff
@@ -72,14 +72,21 @@ def registered_compilers(self) -> Dict[str, CompilerAPI]:
         self._registered_compilers_cache[cache_key] = registered_compilers
         return registered_compilers
 
-    def get_compiler(self, name: str) -> Optional[CompilerAPI]:
+    def get_compiler(self, name: str, settings: Optional[Dict] = None) -> Optional[CompilerAPI]:
         for compiler in self.registered_compilers.values():
-            if compiler.name == name:
-                return compiler
+            if compiler.name != name:
+                continue
+
+            if settings is not None:
+                compiler.compiler_settings = settings
```

Comment:
**issue**: Unable to get multiple compilers of the same type with different settings.

- [ ] Fix